src/views/instances/list/useDeployments.tsx

```diff
@@ -0,0 +1,29 @@
+import { useAuthAxios } from '../../../hooks'
+import { GroupsWithDeployments } from '../../../types'
+import { useState } from 'react'
+
+const useDeployments = () => {
+    const [{ data, error, loading }, refetch] = useAuthAxios<GroupsWithDeployments[]>('/deployments', {
+        useCache: false,
+    })
+    const [showOnlyMyInstances, setShowOnlyMyInstances] = useState(false)
+    const userData = JSON.parse(localStorage.getItem("DHIS2_IM_CURRENT_USER"))
+
+    const filteredData = showOnlyMyInstances
+        ? data?.map(group => ({
```

Comment:
Could we make this whole expression more readable by maybe extracting the deployments first or something? Maybe it's just me but I find it hard to read :slightly_smiling_face:

Reply:
okay, will do.